libs/elastic.go

```diff
@@ -150,6 +150,10 @@ func UpsertUser(ctx context.Context, username string, elasticsearchUser Elastics
 
 	json.NewDecoder(resp.Body).Decode(&body)
 
+	if resp.StatusCode != 200 {
```

Comment:
This prevents that the cache is set if the response code is not 200 which needs to be an error for the outer code to handle it...